cirq/experiments/t2_decay_experiment.py

```diff
@@ -68,10 +69,36 @@ def t2_decay(
     However, during the mid-point of the delay time being measured, a pi-pulse
     (`cirq.X`) gate will be applied to cancel out inhomogeneous dephasing.
     The same method of measuring the final state as Ramsey experiment is applied
-    after the second half of the delay period.
-
-    CPMG, or the Carr-Purcell-Meiboom-Gill sequence, is currently not
-    implemented.
+    after the second half of the delay period.  See the animation on the wiki
+    page https://en.wikipedia.org/wiki/Spin_echo for a visual illustration
+    of this experiment.
+
+    CPMG, or the Carr-Purcell-Meiboom-Gill sequence, involves using a sqrt(Y)
+    followed by a sequence of pi pulses (X gates) in a specific timing pattern:
+        π/2, t, π, 2t, π, ... 2t, π, t
+    The first pulse, a sqrt(Y) gate, will put the qubit's state on the Bloch
+    equator.  After a delay, successive X gates will refocus dehomogenous
+    phase effects by causing them to precess in opposite directions and
+    averaging their effects across the entire pulse train.
+
+    This pulse pattern has two variables that can be adjusted.  The first,
+    denoted as 't' in the above sequence, is delay, which can be specified
+    with `delay_min` and `delaxy_max` or by using a `delay_sweep`, similar to
+    the other experiments.  The second variable is the number of pi pulses
+    (X gates)  This can be specified as a list of integers using the
+    `num_pulses` parameter.  If multiple different pulses are specified,
+    the data will presented in a data frame with two
+    indices (delay_ns and num_pulses).
+
+    See the following reference for more information about CPMG pulse trains:
+    Meiboom, S., and D. Gill, “Modified spin-echo method for measuring nuclear
+    relaxation times”, Rev. Sci. Inst., 29, 688–691 (1958).
+    https://doi.org/10.1063/1.1716296
+
+    Note that interpreting T2 data is fairly tricky and subtle, as it can
+    include other effects that need to be accounted for.  For instance,
+    amplitude damping (T1) will present as T2 noise and needs to be
+    appropriately compensated for to find a true measure of T2.
```

Comment:
Maybe add a remark saying that this is why the class does not actually return a T2 estimate at present.

Reply:
Done I guess.